src/screens/Forum/ForumHomeScreen.js

```diff
@@ -26,56 +33,117 @@ export default class ForumHomeScreen extends React.Component {
       }
     });
 
-    this.unsubscribeFromFirestore = firestore().collection('forum_posts')
-      .orderBy('createdAt', 'desc') /* Update this to order by time approved, most to least recent */
-      .onSnapshot((snapshot) => {
+    const { postLimit } = this.state;
+    firestore().collection('forum_posts')
+      .orderBy('createdAt', 'desc')
+      .limit(postLimit)
+      .get()
+      .then((snapshot) => {
+        // Store data for posts in state
         const forumPosts = snapshot.docs.map((doc) => ({ ...doc.data(), id: doc.id }));
-        const posts = forumPosts.map((post) => {
-          const date = post.createdAt ? post.createdAt.toDate() : null;
-          const time = date ? date.toTimeString() : null;
-          const { currentUserID } = this.state;
+        const lastReferencedPost = forumPosts[forumPosts.length - 1];
 
-          return (
-            <ForumPost
-              belongsToCurrentUser={currentUserID === post.userID}
-              key={post.id}
-              /* Pass in userID  if it exists, other pass in null */
-              userID={post.userID ? post.userID : null}
-              time={time}
-              postID={post.id}
-              navigateToPostScreen={this.navigateToPostScreen}
-            >
-              {post.title}
-            </ForumPost>
-          );
-        });
-        this.setState({ posts, loading: false });
-      }, (error) => { /* Error handler */
-        this.setState({ errorMessage: error.message, loading: false });
-      });
+        this.setState({ forumPosts, lastReferencedPost, loading: false });
+      })
+      .catch((error) => this.setState({ errorMessage: error.message, loading: false }));
   }
 
   componentWillUnmount() {
     this.unsubscribeFromAuth();
-    this.unsubscribeFromFirestore();
   }
 
   navigateToPostScreen(postID, userID) {
     const { navigation } = this.props;
     navigation.navigate('ForumPost', { postID, userID });
   }
 
+  handleEndReached() {
+    const { allPostsLoaded } = this.state;
+
+    if (!allPostsLoaded) {
+      this.setState({ loadingMore: true });
+      this.loadMore();
+    } else { this.setState({ loadingMore: false }); }
```

Comment:
Move `this.setState({ loadingMore: true })` inside `loadMore` and remove the `else` branch.